### DISCUSSION TITLE
Bulk BCR-seq using 10X VDJ Kit, How to run MiXCR pipeline...

### DISCUSSION THREAD

**Discussion by whitebird01** [milaboratory/mixcr#1849] [Q&A]
I recently tried to use MiXCR for analysis, but I encountered a problem. I'm not sure which MiXCR workflow to run and the specific parameter settings.
Let me first describe my problem:
1.The experiment is as follows: Due to the low number of B cells, my colleague wants to analyze the BCR in the samples using Bulk BCR-seq. The reagent used is the 10X VDJ Kit, and the experiment is quite similar to the 10X VDJ protocol, without the GEM generation step ( Gel Beads-in-emulsion) . The sequencing data includes barcode and UMI reads.
2. The workflow tried by MiXCR are:
(1) mixcr analyze 10x-sc-xcr-vdj: I think it is incorrect to use this workflow. The experiment did not generate Gel Beads-in-emulsion and cannot be split according to barcode;
(2) mixcr analyze rna-seq: I first removed the barcode sequence and specified the UMI position.
```
mixcr analyze rna-seq -t 20 \
      --species mmu \
     --tag-pattern "^(UMI:N{10})\\^(R2:*)" \
      R1_Remove_Barcode.fastq.gz  \
      R2.fastq.gz 
```
(3) mixcr analyze generic-amplicon-with-umi: 
```
mixcr analyze generic-amplicon-with-umi  -t 20 \
     --species mmu \
     --rna \
    --tag-pattern "^(CELL:N{16})(UMI:N{10})\\^(R2:*)" \
    --rigid-left-alignment-boundary \
      R1.fastq.gz  \
      R2.fastq.gz 
## or 
mixcr analyze generic-amplicon-with-umi  -t 20 \
     --species mmu \
     --rna \
    --tag-pattern "^(UMI:N{10})\\^(R2:*)" \
    --rigid-left-alignment-boundary \
      R1_Remove_Barcode.fastq.gz  \
      R2.fastq.gz 
```
I have been confused for a long time and would like to ask how I should run the workflow and set parameters based on the current experiments. Thank you so much!

**Comment by mizraelson** [comment]
Hi, sorry for the delay. So can you tell more about your data structure. In 10x the CELL barcode and a UMI barcode usually come from the beads, so I am not sure how did you attach the barcodes without using the beads? Also, did you use the fragmentation step?

**Comment by whitebird01** [comment]
I just saw your reply this afternoon. Thank you for your answer. The experimental procedure involves cDNA fragmentation. The experimental steps are the same as those of conventional 10x V(D)J operations, except for the GEM generation step (Gel Beads-in-Emulsion).
The following is the 10x VDJ library structure:
![image](https://github.com/user-attachments/assets/c24a8a7d-7a56-4fd3-bde9-b1170e07786a)
![vdj library](https://github.com/user-attachments/assets/4192d25a-3de7-48cc-aec5-300c1eb2d1b0)

**Comment by mizraelson** [comment]
Hi, I’ve created a dedicated preset for your type of data. Please find the file attached (unzip it and place it in the folder from which you run MiXCR, or in the `~/.mixcr/presets` directory).

Then run:

```
mixcr analyze local:10x-bulk \
    --species hsa \
    input_R1.fastq.gz \
    input_R2.fastq.gz \
    output
```
Since this is a bulk dataset, I’m using the CELL barcode as an extension of the `^(UMI:N{26})\^(R2:*)`.

Let me know how it works.

[10x-bulk.yaml.zip](https://github.com/user-attachments/files/17709004/10x-bulk.yaml.zip)

**Comment by whitebird01** [comment]
Thank you very much, I will test the process now.

**Reply by racng** [reply]
Hi @whitebird01 , I am interested in using 10x reagent in a bulk reaction as well for troubleshooting purposes, before running 10x with GEM formation. How did your bulk BCR experiment and processing turn out? Do you mind sharing your protocol? Many Thanks!
